src/Protocol/ProtocolRecognizer.cpp

```diff
@@ -238,6 +243,37 @@ void cMultiVersionProtocol::SendDisconnect(cClientHandle & a_Client, const AStri
 
 
 
+bool cMultiVersionProtocol::TryHandleHTTPRequest(cClientHandle & a_Client, ContiguousByteBuffer & a_Data)
+{
+	const auto RedirectUrl = cRoot::Get()->GetServer()->GetCustomRedirectUrl();
+
+	if (RedirectUrl.empty())
+	{
+		return false;
+	}
+
+	ContiguousByteBuffer Buffer;
+	m_Buffer.ReadSome(Buffer, 10U);
+	m_Buffer.ResetRead();
+
+	// The request line, hastily decoded with the hope that it's encoded in US-ASCII.
+	const std::string_view Value(reinterpret_cast<const char *>(Buffer.data()), Buffer.size());
+
+	if (Value == u8"GET / HTTP")
+	{
+		const auto Response = fmt::format(u8"HTTP/1.0 303 See Other\r\nLocation: {}\r\n\r\n", cRoot::Get()->GetServer()->GetCustomRedirectUrl());
```

Comment:
Causes errors on clang 13:
```c++
/home/bond/dev/cuberite/src/Protocol/ProtocolRecognizer.cpp:262:15: error: type of UTF-8 string literal will change from array of const char to array of const char8_t in C++20 [-Werror,-Wc++20-compat]
        if (Value == u8"GET / HTTP")
                     ^
/home/bond/dev/cuberite/src/Protocol/ProtocolRecognizer.cpp:262:15: note: remove 'u8' prefix to avoid a change of behavior; Clang encodes unprefixed narrow string literals as UTF-8
        if (Value == u8"GET / HTTP")
                     ^~
/home/bond/dev/cuberite/src/Protocol/ProtocolRecognizer.cpp:264:37: error: type of UTF-8 string literal will change from array of const char to array of const char8_t in C++20 [-Werror,-Wc++20-compat]
                const auto Response = fmt::format(u8"HTTP/1.0 303 See Other\r\nLocation: {}\r\n\r\n", cRoot::Get()->GetServer()->GetCustomRedirectUrl());
                                                  ^
/home/bond/dev/cuberite/src/Protocol/ProtocolRecognizer.cpp:264:37: note: remove 'u8' prefix to avoid a change of behavior; Clang encodes unprefixed narrow string literals as UTF-8
                const auto Response = fmt::format(u8"HTTP/1.0 303 See Other\r\nLocation: {}\r\n\r\n", cRoot::Get()->GetServer()->GetCustomRedirectUrl());
```

Reply:
Disabled the warning for now since we need the header string to be encoded in ASCII and not a system-native encoding.